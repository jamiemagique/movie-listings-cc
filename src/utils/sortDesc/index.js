import sortDesc from './sortDesc';

export default sortDesc;
