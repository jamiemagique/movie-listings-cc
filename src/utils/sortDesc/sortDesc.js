/**
 * Sorts items in descending order.
 *
 * @param {Object[]} items
 * @param {string} [key="popularity"] - Key to sort on.
 * @returns {Object[]} - Sorted items.
 */
const sortDesc = (items, key = 'popularity') => items.sort((a, b) => b[key] - a[key]);

export default sortDesc;
