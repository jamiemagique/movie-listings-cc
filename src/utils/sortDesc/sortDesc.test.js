import sortDesc from '.';

it('sorts items correctly', () => {
  const items = [
    { id: 1, popularity: 1, },
    { id: 2, popularity: 100, },
    { id: 3, popularity: 50, },
  ];
  expect(sortDesc(items)).toMatchSnapshot();
});
