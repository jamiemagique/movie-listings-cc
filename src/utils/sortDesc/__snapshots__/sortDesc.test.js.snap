// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sorts items correctly 1`] = `
Array [
  Object {
    "id": 2,
    "popularity": 100,
  },
  Object {
    "id": 3,
    "popularity": 50,
  },
  Object {
    "id": 1,
    "popularity": 1,
  },
]
`;
